### ISSUE TITLE
custom index_analyzer not found

### ISSUE THREAD

**Issue by dmaniloff** [elastic/elasticsearch-dsl-py#180]
Hello, thank you for `elasticsearch-dsl`!

I am writing out a simple example with custom analyzers. Everything works except for the index_analyzer kwarg not accepting a custom analyzer. I've tried several variations, also using the Mapping class, and still get the same error. Here's my setup:

```
shingle_analyzer = dsl.analyzer(
    'shingle_analyzer',
    type='custom',
    tokenizer='standard',
    filter=[
        'lowercase',
        dsl.token_filter(
            'shingle_filter', type='shingle',
            min_shingle_size=2, max_shingle_size=2
        )
    ]
)

ngram_analyzer = dsl.analyzer(
    'ngram_analyzer',
    type='custom',
    tokenizer='standard',
    filter=[
        'lowercase',
        dsl.token_filter(
            'ngram_filter', type='nGram',
            min_gram=1, max_gram=20
        )
    ]
)

edge_ngram_analyzer = dsl.analyzer(
    'edge_ngram_analyzer',
    type='custom',
    tokenizer='standard',
    filter=[
        'lowercase',
        dsl.token_filter(
            'edge_ngram_filter', type='edgeNGram',
            min_gram=1, max_gram=20
        )
    ]
)

class Product(dsl.DocType):
    name = dsl.String(
        analyzer='english',
        fields={
            'std': dsl.String(analyzer='standard'),
            'raw': dsl.String(index='not_analyzed'),
            'shingles': dsl.String(analyzer=shingle_analyzer),
            'inautocomplete': dsl.String(
                index_analyzer=ngram_analyzer,
                search_analyzer='standard',
            ),
            'edgeautocomplete': dsl.String(
                index_analyzer=edge_ngram_analyzer,
                search_analyzer='standard',
            ),
        },
    )
    tags = dsl.String(index='not_analyzed')
    categories = dsl.String(index='not_analyzed')

    class Meta:
        index = 'products_v7'

```

Both `index_analyzer=ngram_analyzer` and ` index_analyzer=edge_ngram_analyzer` are causing this error:

```
RequestError: TransportError(400, u'MapperParsingException[mapping [product]]; nested: MapperParsingException[Analyzer [ngram_analyzer] not found for field [inautocomplete]]; ')
```

I am using 

```
{
   "status": 200,
   "name": "Chamber",
   "cluster_name": "elasticsearch",
   "version": {
      "number": "1.6.0",
      "build_hash": "cdd3ac4dde4f69524ec0a14de3828cb95bbb86d0",
      "build_timestamp": "2015-06-09T13:36:34Z",
      "build_snapshot": false,
      "lucene_version": "4.10.4"
   },
   "tagline": "You Know, for Search"
}
```
and `elasticsearch-dsl==0.0.4` 

Thanks in advance.

**Comment by HonzaKral** [comment]
Thanks for the report!

I modified one of the tests to use `index_` and `search_` analyzers in aae32e688e57e23a4cf16abe8288bb3fe29766f2 but cannot seem to replicate your problem as they get collected properly.

Could you share where you get this error, how are you creating the mappings in es from your `DocType` classes?

**Comment by HonzaKral** [comment]
I cannot replicate this issue so without further information I will close it.

**Comment by dmaniloff** [comment]
solved in 0.0.8.
